conans/test/integration/cmake_flags_test.py

```diff
@@ -201,7 +201,8 @@ def build(self):
         client.save({"conanfile.py": conanfile % "True"})
         client.run("build", ignore_error=True)
 
-        self.assertIn("Execute 'conan install -g txt' first", client.user_io.out)
+        self.assertIn("self.deps_cpp_info not defined. If you need it for a local command run "
```

Comment:
I removed from the ``load_consumer_conanfile`` the:

```
if not profile:
    raise ConanException("Execute 'conan install -g txt' first.")
```

Because now it always receives the default profile (from the `self.client_cache.default_profile`). I unified the instantiation of the loader, there was a big mess of different instantiations with different parameters depending on the call. 
So I think it has the sense to get always the default profile because it always applies, right?

So, now, this test fails a little bit later, in `_load_deps_cpp_info`